packages/client/test/rpc/engine/preimages.spec.ts

```diff
@@ -0,0 +1,258 @@
+        receiptTrie: '0xd95b673818fa493deec414e01e610d97ee287c9421c8eff4102b1647c1a184e4',
+        gasUsed: '0xa410',
+        coinbase: '0x9da2abca45e494476a21c49982619ee038b68556',
+      },
+      // add preimages for addresses accessed in txs
+      preImages: [
+        // coinbase
+        '0x9da2abca45e494476a21c49982619ee038b68556',
+        // withdrawals
+        '0x2000000000000000000000000000000000000000',
+        '0x2100000000000000000000000000000000000000',
+        '0x2200000000000000000000000000000000000000',
+        '0x2300000000000000000000000000000000000000',
+        '0x2400000000000000000000000000000000000000',
+        '0x2500000000000000000000000000000000000000',
+        '0x2600000000000000000000000000000000000000',
+        '0x2700000000000000000000000000000000000000',
+        // txs
+      ],
+    },
+    {
+      name: 'block 3 no txs with just withdrawals but zero coinbase',
+      blockData: {
+        transactions: [] as string[],
+        blockNumber: '0x03',
+        stateRoot: '0xe4538f9d7531eb76e82edf7480e4578bc2be5f454ab02db4d9db6187dfa1f9ca',
+        receiptTrie: '0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421',
+        gasUsed: '0x0',
+        coinbase: '0x0000000000000000000000000000000000000000',
+      },
+      preImages: [
+        // coinbase
+        '0x0000000000000000000000000000000000000000',
+        // withdrawals,
+        '0x3000000000000000000000000000000000000000',
+        '0x3100000000000000000000000000000000000000',
+        '0x3200000000000000000000000000000000000000',
+        '0x3300000000000000000000000000000000000000',
+        '0x3400000000000000000000000000000000000000',
+        '0x3500000000000000000000000000000000000000',
+        '0x3600000000000000000000000000000000000000',
+        '0x3700000000000000000000000000000000000000',
+        // no txs
+      ],
+    },
+    {
+      name: 'block 4 no txs with just withdrawals',
+      blockData: {
+        transactions: blocks.block13.execute.transactions,
+        blockNumber: '0x04',
+        stateRoot: '0x57e675e1d6b2ab5d65601e81658de1468afad77752a271a48364dcefda856614',
+        receiptTrie: '0x6a0be0e8208f625225e43681258eb9901ed753e2656f0cd6c0a3971fada5f190',
+        gasUsed: '0x3c138',
+        coinbase: '0xa874386cdb13f6cb3b974d1097b25116e67fc21e',
+      },
+      preImages: [
+        // coinbase
+        '0xa874386cdb13f6cb3b974d1097b25116e67fc21e',
+        // withdrawals
+        '0x4000000000000000000000000000000000000000',
+        '0x4100000000000000000000000000000000000000',
+        '0x4200000000000000000000000000000000000000',
+        '0x4300000000000000000000000000000000000000',
+        '0x4400000000000000000000000000000000000000',
+        '0x4500000000000000000000000000000000000000',
+        '0x4600000000000000000000000000000000000000',
+        '0x4700000000000000000000000000000000000000',
+      ],
+      // no txs
```

Comment:
Are there no txs in this block? It seems that there are?

Reply:
Should there also be a test where a contract is invoked and a storage slot is written to?

Reply:
@g11tech would know more about those cases than me, would you mind clarifying gajinder?

Reply:
yes there are, we need to run the test with debug enabled for evm and add the list of things accessed here

Reply:
I logged the accesses and added some tx preimages to the two blocks that have transactions.